wgpu/src/lib.rs

```diff
@@ -388,8 +388,9 @@ static_assertions::assert_impl_all!(SurfaceConfiguration: Send, Sync);
 /// [`GPUCanvasContext`](https://gpuweb.github.io/gpuweb/#canvas-context)
 /// serves a similar role.
 #[derive(Debug)]
-pub struct Surface {
+pub struct Surface<'window> {
     context: Arc<C>,
+    surface: PhantomData<&'window ()>,
```

Comment:
My assumption here was that `Surface` is the only type requiring the lifetime, e.g. `SurfaceTexture` doesn't. Is this correct?

Reply:
Trying to render to the surface after it's destroyed does seem like a questionable operation. I was wondering if `Surface` could return a `&'window SurfaceTexture`.

Reply:
We do have visibility in wgpu-core if there are any active surface textures - so we should error on surface unconfigure, reconfigure, or surface drop.

Reply:
Actually, not that I think about it, if this is really an issue, it was an issue already before this PR.